### PR TITLE
CNV-39605: route creation form

### PR DESCRIPTION
**Pull request by upalatucci**
image

### CODE REVIEW

File: src/views/routes/form/RouteForm.tsx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+import TLSTermination from './TLSTermination';
+
+type RouteFormProps = {};
+
+const RouteForm: FC<RouteFormProps> = () => {
+  const { t } = useNetworkingTranslation();
+  const navigate = useNavigate();
+  const [apiError, setError] = useState<Error>(null);
+  const [activeNamespace] = useActiveNamespace();
+  const namespace = ALL_NAMESPACES_KEY === activeNamespace ? DEFAULT_NAMESPACE : activeNamespace;
```
Comment (pcbailey) [inline src/views/routes/form/RouteForm.tsx:42]: I recommend adding a function in src/utils like the one we have in the kubevirt-plugin for this.

```
export const getValidNamespace = (activeNamespace: string) =>
  activeNamespace === ALL_NAMESPACES_SESSION_KEY ? DEFAULT_NAMESPACE : activeNamespace;
```

File: src/views/routes/form/ServiceSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            toggle={(toggleRef: Ref<MenuToggleElement>) => (
+              <MenuToggle
+                id="toggle-route-service"
+                isExpanded={isDropdownOpen}
+                isFullWidth
+                onClick={() => setIsDropdownOpen(!isDropdownOpen)}
+                ref={toggleRef}
+              >
+                {value ? (
+                  <>
```
Comment (pcbailey) [inline src/views/routes/form/ServiceSelector.tsx:67]: Are these necessary?

File: src/views/routes/form/ServiceSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                    <ResourceIcon groupVersionKind={ServiceGroupVersionKind} /> {value}
+                  </>
+                ) : (
+                  t('Select a Service')
+                )}
+              </MenuToggle>
+            )}
+          >
+            <DropdownList>
+              {services.map((service) => (
```
Comment (pcbailey) [inline src/views/routes/form/ServiceSelector.tsx:77]: Should we add a conditional chaining operator here?

File: src/views/routes/form/TLSTermination.tsx [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+  control: Control<RouteKind, any>;
+  watch: UseFormWatch<RouteKind>;
+};
+
+const TLSTermination: FC<TLSTerminationProps> = ({ control, watch }) => {
+  const { t } = useNetworkingTranslation();
+  const [isTerminationOpen, setIsTerminationOpen] = useState(false);
+  const [isInsecureTrafficOpen, setIsInsecureTrafficOpen] = useState(false);
+
+  const isPassthrought = watch('spec.tls.termination') === PASSTHROUGH;
```
Comment (pcbailey) [inline src/views/routes/form/TLSTermination.tsx:31]: Type: isPassthrought -> isPassthrough

File: src/views/routes/form/ServiceSelector.tsx (hunk 3 above)
Comment (upalatucci) [reply src/views/routes/form/ServiceSelector.tsx:77]: okay i'll do something

File: src/views/routes/form/AlternateService.tsx [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+const AlternateService: FC<AlternateServiceProps> = ({ services }) => {
+  const { t } = useNetworkingTranslation();
+
+  const { control, register, watch } = useFormContext<RouteKind>();
+
+  const { append, fields, remove } = useFieldArray({ control, name: 'spec.alternateBackends' });
+
+  const selectedServiceNames = watch('spec.alternateBackends')?.map((field) => field.name) || [];
+  const filtedServices = (services || []).filter(
```
Comment (pcbailey) [inline src/views/routes/form/AlternateService.tsx:35]: Typo: filtedServices -> filteredServices

File: src/views/routes/form/AlternateService.tsx [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+    (service) => !selectedServiceNames.includes(getName(service)),
+  );
+
+  return (
+    <>
+      {fields.map((field, index) => (
+        <Controller
```
Comment (pcbailey) [inline src/views/routes/form/AlternateService.tsx:42]: Suggestion: Extract this out into a separate component.

File: src/views/routes/details/components/tabs/detailsTab/RouteDetailsTab.tsx [hunk 7]
```diff
@@ -1,10 +1,10 @@
 import React, { FC } from 'react';
 import * as _ from 'lodash';
```
Comment (metalice) [inline src/views/routes/details/components/tabs/detailsTab/RouteDetailsTab.tsx:2]: can get rid of lodash?

File: src/views/routes/details/components/tabs/detailsTab/components/RouteIngressStatusSection/IngressStatusSection.tsx [hunk 8]
```diff
@@ -1,8 +1,8 @@
 import React, { FC } from 'react';
 import * as _ from 'lodash';
```
Comment (metalice) [inline src/views/routes/details/components/tabs/detailsTab/components/RouteIngressStatusSection/IngressStatusSection.tsx:2]: can get rid of lodash?

File: src/views/routes/form/RouteForm.tsx [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+    control,
+    formState: { isSubmitting, isValid },
+    handleSubmit,
+    register,
+    watch,
+  } = methods;
+
+  const tls = watch('spec.tls');
+
+  const isSecured = tls !== null && tls !== undefined;
```
Comment (metalice) [inline src/views/routes/form/RouteForm.tsx:69]: simply tls ?

File: src/views/routes/form/RouteForm.tsx [hunk 10]
```diff
@@ -0,0 +1,181 @@
+import React, { FC, useState } from 'react';
```
Comment (metalice) [inline src/views/routes/form/RouteForm.tsx:1]: please see how u can split this file

File: src/views/routes/form/ServiceSelector.tsx (hunk 2 above)
Comment (metalice) [reply src/views/routes/form/ServiceSelector.tsx:60]: fair point

File: src/views/routes/form/ServiceSelector.tsx [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+
+  if (error) return <Alert title={t('Error')}></Alert>;
+
+  return (
+    <>
+      <Controller
+        control={control}
+        name="spec.to.name"
+        render={({ field: { onChange, value } }) => (
+          <FormGroup fieldId="service" isRequired label={t('Service')}>
```
Comment (metalice) [inline src/views/routes/form/ServiceSelector.tsx:54]: please export all fieldID'sall across this p to const file

**Comment by metalice** [comment]
please go overall the files in the pr and remove lodash where it used thank you

File: src/views/routes/form/ServiceSelector.tsx (hunk 2 above)
Comment (upalatucci) [reply src/views/routes/form/ServiceSelector.tsx:60]: as you can see we have 

```
   <>
     <Component /> {text}
   </>
 
```

**Comment by upalatucci** [comment]
@metalice i removed some reference to lodash but i think we need a different pr to remove all of them

File: src/views/routes/form/AlternateServicesSection.tsx [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+      {fields.map((field, index) => (
+        <AlternateService
+          field={field}
+          filteredServices={filteredServices}
+          index={index}
+          key={field.id}
+        />
+      ))}
+
+      {filteredServices?.length !== 0 && (
```
Comment (metalice) [inline src/views/routes/form/AlternateServicesSection.tsx:40]: !isEmpty

File: src/views/routes/form/AlternateServicesSection.tsx [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+        <Button
+          className="pf-m-link--align-left co-create-route__add-service-btn"
+          isInline
+          onClick={() => append({ kind: 'Service', name: '', weight: 100 })}
+          type="button"
+          variant="link"
```
Comment (metalice) [inline src/views/routes/form/AlternateServicesSection.tsx:46]: ButtonVariant

**Pull request merged**